unison-core/src/Unison/Names2.hs

```diff
@@ -216,6 +217,26 @@ addType n r = (<> fromTypes [(n, r)])
 addTerm :: Ord n => n -> Referent -> Names' n -> Names' n
 addTerm n r = (<> fromTerms [(n, r)])
 
+-- | Like hqTermName and hqTypeName, but considers term and type names to
+-- conflict with each other (so will hash-qualify if there is e.g. both a term
+-- and a type named "foo").
+--
+-- This is useful in contexts such as printing branch diffs. Example:
+--
+--     - Deletes:
+--
+--       foo
+--       foo
+--
+-- We want to append the hash regardless of whether or not one is a term and the
+-- other is a type.
```

Comment:
No, I think we want:
```
   - Deletes:

     foo (Type)
     foo (Term)
```
rather than hashes, right?  Actually, probably all of the deletes should indicate or be grouped by term/type.

Reply:
I'm also okay with merging this as-is but with a separate ticket to track this— this would be the first instance where hash-qualifying is used as substitute for distinguishing Term from Type, which I think is the wrong direction.

Reply:
Thinking it over some more, I dunno.  Hash-qualified is an improvement over indistinguishable "name-only"s, so this is fine for now; I dunno what the answer is in the long term.  My guess is it's "not have multiple parallel namespaces".

Reply:
I agree having separate "Deleted terms" and "Deleted types" stanzas would improve this situation. The current pretty-printer actually just space-separates things, so this might look a bit weird:

```
- Deletes:

    foo (Type) foo (Term) bar
```

Reply:
I see, yeah.

Reply:
Let's just — I dunno whether in a ticket or in comments in some appropriate place — remember that this is something we might be able to improve later.